include/squid.h

```diff
@@ -61,5 +61,11 @@ using namespace Squid;
 
 #define LOCAL_ARRAY(type, name, size) static type name[size]
 
+#if USE_OPENSSL
+#if _SQUID_FREEBSD_
+#define OPENSSL_KTLS_SUPPORT 1
+#endif
+#endif
+
```

Comment:
PR description indicates there is a `./configure` option added. But no logic change adds that option.

Either the option should be added and this chunk dropped entirely.
OR this chunk moved to `compat/os/freebsd.h` since the change is OS-specific to FreeBSD, with PR description updated to correctly state what is being added.

Reply:
Hi.

My first intention in this PR was to add the configurable option "--with-openssl-ktls" to enable "USE_OPENSSL_KTLS" (for every session), so I had changed configure.ac etc.:
 https://github.com/squid-cache/squid/pull/1629/commits/1f7e2da4cebed8a32f101245facfbe21689d01fa

However, rousskov suggested me, if logically possible, that not adding the build option was desireble, and instead it should be just configurable (e.g. http_port and tls_outgoing_options):
 https://github.com/squid-cache/squid/pull/1629#discussion_r1443285736

So, I cancelled (reverted) configure.ac or related changes, and introduced OPENSSL_KTLS_SUPPORT (which is not user-configurable but OS-dependent) to separate codes for KTLS enabled or disabled platforms.


Regarding "OS-specific to FreeBSD," it's because I can test only on the platform (at least now).
Linux, for example, may also be able to support (can define OPENSSL_KTLS_SUPPORT) with or without modifications of code in the future, if some one could test it.
Even though, if changing the place to define OPENSSL_KTLS_SUPPORT or renaming it is desired, there is no reason not to do so.